File: code_examples/3-Structured_Problem_Solving/5-Example_Sequence/CPP/main.cpp
// Copyright (c) 2020 St. Mother Teresa HS All rights reserved.
//
// Created by: Mr. Coxall
// Created on: Sep 2020
// This program calculates total from subtotal and tax

#include <iostream>
#include <iomanip>

int main() {
    // this function calculates total from subtotal and tax
    const float HST = 0.13;
    float tax;
    float subTotal;
    float total;

    // input
    std::cout << "Enter the subtotal: $";
    std::cin >> subTotal;

    // process
    tax = + subTotal * HST;
    total = subTotal + tax;

    // output
    std::cout << "" << std::endl;
    std::cout << "The HST is: $" 
                << std::fixed << std::setprecision(2) << std::setfill('0')
                << tax << " and the total cost is: $"
                << std::fixed << std::setprecision(2) << std::setfill('0')
                << total << "." << std::endl;

    std::cout << "\nDone. " << std::endl;
}

File: code_examples/3-Structured_Problem_Solving/5-Example_Sequence/Python/main.py
#!/usr/bin/env python3
"""
Created by: Mr. Coxall
Created on: Sep 2020
This module calculates total from subtotal and tax
"""

import constants


def main() -> None:
    """The main() function calculates total from subtotal and tax, returns None."""

    # input
    sub_total = float(input("Enter the subtotal: $"))

    # process
    tax = sub_total * constants.HST
    total = sub_total + tax

    # output
    print("")
    print(f"The HST is ${tax:,.2f}.")
    print(f"The total cost is ${total:,.2f}.")

    print("\nDone.")


if __name__ == "__main__":
    main()
